pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alicloudbotanist
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/operation/terraformer"
+)
+
+type cloudConfig struct {
+	Global struct {
+		KubernetesClusterTag string
+		UID                  string `json:"uid"`
+		VpcID                string `json:"vpcid"`
+		Region               string `json:"region"`
+		ZoneID               string `json:"zoneid"`
+		VswitchID            string `json:"vswitchid"`
+
+		AccessKeyID     string `json:"accessKeyID"`
+		AccessKeySecret string `json:"accessKeySecret"`
+	}
+}
+
+// GenerateCloudProviderConfig generates the Alicloud cloud provider config.
+// See this for more details:
+// https://github.com/kubernetes/cloud-provider-alibaba-cloud/blob/master/cloud-controller-manager/alicloud.go#L62
+func (b *AlicloudBotanist) GenerateCloudProviderConfig() (string, error) {
+	var (
+		vpcID     = "vpc_id"
+		vswitchID = fmt.Sprintf("vswitch_id_z%d", 0)
+	)
+	tf, err := terraformer.NewFromOperation(b.Operation, common.TerraformerPurposeInfra)
+	if err != nil {
+		return "", err
+	}
+	stateVariables, err := tf.GetStateOutputVariables(vpcID, vswitchID)
+	if err != nil {
+		return "", err
+	}
+
+	key := base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeyID])
+	secret := base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeySecret])
+	cfg := `{
```

Comment:
You already have defined the Go type (https://github.com/gardener/gardener/pull/626/files#diff-bf860fa04669b8811da4ea237535b295R26) - why not using it here and doing this cleanly?

Reply:
Done.